packages/api/test/e2e/doughnut-v0.e2e.ts

```diff
@@ -0,0 +1,184 @@
+// Copyright 2019 Centrality Investments Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import testingPairs from '@plugnet/keyring/testingPairs';
+import { Doughnut } from '@cennznet/types/extrinsic/Doughnut';
+import { hexToU8a, assert } from '@plugnet/util'
+import { encode as encodeCennznut } from '@cennznet/cennznut';
+import { generate as encodeDoughnut } from '@plugnet/doughnut-maker';
+
+import { Api } from '../../src/Api';
+import { Keypair } from '@plugnet/util-crypto/types';
+import { KeyringPair } from '@cennznet/util/types';
+import { Extrinsic } from '@cennznet/types/extrinsic';
+
+/// Helper for creating CENNZnuts
+function makeCennznut(module: string, method: string): Uint8Array {
+  return encodeCennznut(0, {
+    "modules": {
+      [module]: {
+        "methods": {
+          [method]: {}
+        }
+      }
+    }
+  });
+}
+
+/// Helper for creating v0 Doughnuts
+function makeDoughnut(
+  issuer: Keypair,
+  holder: KeyringPair,
+  permissions: Record<string, Uint8Array>,
+): Uint8Array {
+  return encodeDoughnut(
+    0,
+    0,
+    {
+      issuer: issuer.publicKey,
+      holder: holder.publicKey,
+      expiry: 55555,
+      block_cooldown: 0,
+      permissions: permissions,
+    },
+    issuer
+  );
+}
+
+describe('Doughnut for CennznetExtrinsic', () => {
+  let aliceKeyPair = {
+    secretKey: hexToU8a('0x98319d4ff8a9508c4bb0cf0b5a78d760a0b2082c02775e6e82370816fedfff48925a225d97aa00682d6a59b95b18780c10d7032336e88f3442b42361f4a66011'),
+    publicKey: hexToU8a('0xd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d')
+  };
+  let api: Api;
+  let keyring;
+
+  beforeAll(async () => {
+    // TODO: Update to Rimu
+    api = await Api.create({ provider: 'ws://localhost:9944' });
+    keyring = testingPairs({ type: 'sr25519' });
+  });
+
+  afterEach(() => {
+    jest.setTimeout(5000);
+  });
+
+  it('Delegates a GA transfer from alice to charlie when extrinsic is signed by bob', async done => {
```

Comment:
i got an error when ran the test, `@polkadot/wasm-crypto has not been initialized`. can you switch `doughnut-maker` to use `@plugnet/wasm-crypto` which is a fork of `@polkadot/wasm-crypto` and used in `@cennznet/*`

Reply:
Strange, I don't have this error. Can this be awaited in the test setup instead?

Reply:
After rethinking about that, i think it's fine for doughnut-maker to use whatever crypto lib. but since doughnut-maker depends on `@polkadot/wasm-crypto`'s initialization, it makes more sense if you can re-export a `waitReady()` from doughnut-maker.

Reply:
Doughnuts aren't a plugnet technology, and pulling in a plugnet scoped package invites future bloat. The doughnut packages should be lightweight.

While the doughnut package is currently published under plugnet, that's likely to change in the near future. Doughnuts should be scoped to centrality directly.

But this is a separate concern to the technical issue. If there is another way to work around the parity bug then please go in that direction.

@holygits, we ran into the issue during the hackathon, and switching to `@plugnet/wasm-crypto`'s special init method didn't fix the issue. Feel free to dive deeper into it, and test the `@plugnet/wasm-crypto` to get a feel for the issue.

Reply:
@ianhe8x can you give steps to reproduce the await failure? I did a clean clone and ran `npx lerna bootstrap; npx jest -c jest/jest.config.js packages/api/test/e2e/doughnut-v0.e2e.ts`

Reply:
you will get it by removing Api.create in beforeAll. You don't have this error is simply because it takes less time than Api.create to initialize, so when the test code hit the line, it's already ready.

Reply:
if you want this requirement transparent to end user, i suggest change makeDoughnut() to async, then internally you can await wasm-crypto ready before continue. (which also make it friendly to external signer)